tests/nvmeof/workflows/nvme.py

```diff
@@ -0,0 +1,435 @@
+                cfg["config"]["specs"][0]["spec"]["group"] = gw_group
+            else:
+                cfg["config"]["pos_args"].append(gw_group)
+
+            # Add rebalance period if specified
+            if config.get("rebalance_period", False):
+                rebalance_sec = config.get("rebalance_period_sec", 0)
+                cfg["config"]["specs"][0]["spec"][
+                    "rebalance_period_sec"
+                ] = rebalance_sec
+
+            return cfg
+
+        return cfg
+
+    def _create_spec_deployment_config(
+        self, config, rbd_pool, gw_nodes, gw_group, use_encryption
+    ):
+        """Create spec-based deployment configuration."""
+        spec = {
+            "service_type": "nvmeof",
+            "service_id": rbd_pool,
+            "mtls": config.get("mtls", False),
+            "placement": self._get_placement_config(config, gw_nodes),
+            "spec": {
+                "pool": rbd_pool,
+                "enable_auth": config.get("mtls", False),
+            },
+        }
+
+        # Add encryption if specified
+        if use_encryption:
+            spec["encryption"] = True
+
+        # Add group if specified
+        if gw_group:
+            spec["spec"]["group"] = gw_group
+
+        cfg = {
+            "no_cluster_state": False,
+            "config": {
+                "command": "apply_spec",
+                "service": "nvmeof",
+                "validate-spec-services": config.get("validate-spec-services", True),
+                "specs": [spec],
+            },
+        }
+
+        return cfg
+
+    def _create_apply_deployment_config(self, config, rbd_pool, gw_nodes, gw_group):
+        """Create apply-based deployment configuration."""
+        cfg = {
+            "no_cluster_state": False,
+            "config": {
+                "command": "apply",
+                "service": "nvmeof",
+                "args": {"placement": self._get_placement_config(config, gw_nodes)},
+                "pos_args": [rbd_pool],
+            },
+        }
+
+        return cfg
+
+    def _get_placement_config(self, config, gw_nodes):
+        """Get placement configuration based on config options."""
+        placement = {"nodes": [i.hostname for i in gw_nodes]}
+
+        # Add label-based placement if specified
+        if config.get("label"):
+            placement["label"] = config["label"]
+
+        # Add limit if specified
+        if config.get("limit"):
+            placement["limit"] = config["limit"]
+
+        # Add separator if specified
+        if config.get("sep"):
+            placement["sep"] = config["sep"]
+
+        return placement
+
+    def _get_deploy_config(self):
+        """
+        Prepare config for test_nvmeof.run based on self.config.
+        """
+        # This is a direct copy of the deploy_nvme_service logic
+        from tests.nvmeof.workflows.nvme_utils import apply_nvme_sdk_cli_support
```

Comment:
Let call it outside the class at top of the module.
```suggestion
```

Reply:
I have removed this method as of now. will add it if required.